pkg/kubelet/pod_workers.go

```diff
@@ -981,26 +981,30 @@ func calculateEffectiveGracePeriod(status *podSyncStatus, pod *v1.Pod, options *
 	// enforce the restriction that a grace period can only decrease and track whatever our value is,
 	// then ensure a calculated value is passed down to lower levels
 	gracePeriod := status.gracePeriod
-	// this value is bedrock truth - the apiserver owns telling us this value calculated by apiserver
-	if override := pod.DeletionGracePeriodSeconds; override != nil {
-		if gracePeriod == 0 || *override < gracePeriod {
+
```

Comment:
Can you explain why you are making this change?  Spec is not the most correct value, it is overridden by the value in metadata?

Reply:
If `gracePeriod == 0 || ` is still existing, when I execute `kubectl delete pod --force --grace-period=0`, then return value of calculateEffectiveGracePeriod() become `pod.Spec.TerminationGracePeriodSeconds` (default: 300) because `gracePeriod` value is 0 and go through this(L.986) if statement.

Reply:
Actually, the kubelet biggest problem with this bug is the inability to send the second deletion message to the CRI